### ISSUE TITLE
GitHub error 502 Bad Gateway

### ISSUE THREAD

**Issue by bookmoons** [MyBitFoundation/github-proxy#21]
A local run shows `502 Bad Gateway` errors for every GitHub request. Running with the `GITHUB_TOKEN` envvar set to a personal token.

I'm not sure where to go with a 5xx error. Does anyone have an idea what might be causing this?

```
{ Error: Request failed with status code 502
    at createError (/home/user/repo/github/bookmoons/github-proxy/node_modules/axios/lib/core/createError.js:16:15)
    at settle (/home/user/repo/github/bookmoons/github-proxy/node_modules/axios/lib/core/settle.js:18:12)
    at IncomingMessage.handleStreamEnd (/home/user/repo/github/bookmoons/github-proxy/node_modules/axios/lib/adapters/http.js:201:11)
    at IncomingMessage.emit (events.js:187:15)
    at endReadableNT (_stream_readable.js:1094:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
  config:
   { adapter: [Function: httpAdapter],
     transformRequest: { '0': [Function: transformRequest] },
     transformResponse: { '0': [Function: transformResponse] },
     timeout: 0,
     xsrfCookieName: 'XSRF-TOKEN',
     xsrfHeaderName: 'X-XSRF-TOKEN',
     maxContentLength: -1,
     validateStatus: [Function: validateStatus],
     headers:
      { Accept: 'application/json, text/plain, */*',
        'Content-Type': 'application/json;charset=utf-8',
        Authorization: 'token ELIDED',
        'User-Agent': 'axios/0.18.0',
        'Content-Length': 2041 },
     method: 'post',
     url: 'https://api.github.com/graphql',
     data:
      '{"query":"\\nquery {\\n  organization(login: \\"MyBitFoundation\\"){\\n    repositories(first: 100){\\n      pageInfo{\\n        hasNextPage\\n      }\\n      edges{\\n        node{\\n          name,\\n          repositoryTopics(first: 10){\\n            edges{\\n              node{\\n                topic{\\n                  name\\n                }\\n              }\\n            }\\n          },\\n          issues(first: 100){\\n            pageInfo{\\n              hasNextPage\\n            }\\n            edges{\\n              node{\\n                createdAt,\\n                state,\\n                timeline(first: 10){\\n                  pageInfo{\\n                    hasNextPage\\n                  }\\n                  edges{\\n                    node{\\n                      ... on CrossReferencedEvent{\\n                        url\\n                        target\\n                        source{\\n                          ... on PullRequest{\\n                            url\\n                            state\\n                            author{\\n                              login\\n                            }\\n                          }\\n                        }\\n                      }\\n                    }\\n                    cursor\\n                  }\\n                },\\n                labels(first: 10){\\n                  edges{\\n                    node{\\n                      name,\\n                    },\\n                  }\\n                },\\n                url,\\n                title,\\n                number\\n                comments(first: 10){\\n                  pageInfo{\\n                    hasNextPage\\n                  }\\n                  edges{\\n                    node{\\n                      body,\\n                      author{\\n                        login\\n                      }\\n                    },\\n                    cursor\\n                  }\\n                }\\n              },\\n              cursor\\n            }\\n          }\\n        },\\n        cursor\\n      }\\n    }\\n  }\\n}"}' },
  request:
   ClientRequest {
     _events:
      { socket: [Function],
        abort: [Function],
        aborted: [Function],
        error: [Function],
        timeout: [Function],
        prefinish: [Function: requestOnPrefinish] },
     _eventsCount: 6,
     _maxListeners: undefined,
     output: [],
     outputEncodings: [],
     outputCallbacks: [],
     outputSize: 0,
     writable: true,
     _last: true,
     chunkedEncoding: false,
     shouldKeepAlive: false,
     useChunkedEncodingByDefault: true,
     sendDate: false,
     _removedConnection: false,
     _removedContLen: false,
     _removedTE: false,
     _contentLength: null,
     _hasBody: true,
     _trailer: '',
     finished: true,
     _headerSent: true,
     socket:
      TLSSocket {
        _tlsOptions: [Object],
        _secureEstablished: true,
        _securePending: false,
        _newSessionPending: false,
        _controlReleased: true,
        _SNICallback: null,
        servername: 'api.github.com',
        alpnProtocol: false,
        authorized: true,
        authorizationError: null,
        encrypted: true,
        _events: [Object],
        _eventsCount: 8,
        connecting: false,
        _hadError: false,
        _handle: [TLSWrap],
        _parent: null,
        _host: 'api.github.com',
        _readableState: [ReadableState],
        readable: true,
        _maxListeners: undefined,
        _writableState: [WritableState],
        writable: false,
        allowHalfOpen: false,
        _sockname: null,
        _pendingData: null,
        _pendingEncoding: '',
        server: undefined,
        _server: null,
        ssl: [TLSWrap],
        _requestCert: true,
        _rejectUnauthorized: true,
        parser: null,
        _httpMessage: [Circular],
        [Symbol(res)]: [TLSWrap],
        [Symbol(asyncId)]: 4112,
        [Symbol(lastWriteQueueSize)]: 0,
        [Symbol(timeout)]: null,
        [Symbol(kBytesRead)]: 0,
        [Symbol(kBytesWritten)]: 0,
        [Symbol(connect-options)]: [Object] },
     connection:
      TLSSocket {
        _tlsOptions: [Object],
        _secureEstablished: true,
        _securePending: false,
        _newSessionPending: false,
        _controlReleased: true,
        _SNICallback: null,
        servername: 'api.github.com',
        alpnProtocol: false,
        authorized: true,
        authorizationError: null,
        encrypted: true,
        _events: [Object],
        _eventsCount: 8,
        connecting: false,
        _hadError: false,
        _handle: [TLSWrap],
        _parent: null,
        _host: 'api.github.com',
        _readableState: [ReadableState],
        readable: true,
        _maxListeners: undefined,
        _writableState: [WritableState],
        writable: false,
        allowHalfOpen: false,
        _sockname: null,
        _pendingData: null,
        _pendingEncoding: '',
        server: undefined,
        _server: null,
        ssl: [TLSWrap],
        _requestCert: true,
        _rejectUnauthorized: true,
        parser: null,
        _httpMessage: [Circular],
        [Symbol(res)]: [TLSWrap],
        [Symbol(asyncId)]: 4112,
        [Symbol(lastWriteQueueSize)]: 0,
        [Symbol(timeout)]: null,
        [Symbol(kBytesRead)]: 0,
        [Symbol(kBytesWritten)]: 0,
        [Symbol(connect-options)]: [Object] },
     _header:
      'POST /graphql HTTP/1.1\r\nAccept: application/json, text/plain, */*\r\nContent-Type: application/json;charset=utf-8\r\nAuthorization: token ELIDED\r\nUser-Agent: axios/0.18.0\r\nContent-Length: 2041\r\nHost: api.github.com\r\nConnection: close\r\n\r\n',
     _onPendingData: [Function: noopPendingOutput],
     agent:
      Agent {
        _events: [Object],
        _eventsCount: 1,
        _maxListeners: undefined,
        defaultPort: 443,
        protocol: 'https:',
        options: [Object],
        requests: {},
        sockets: [Object],
        freeSockets: {},
        keepAliveMsecs: 1000,
        keepAlive: false,
        maxSockets: Infinity,
        maxFreeSockets: 256,
        maxCachedSessions: 100,
        _sessionCache: [Object] },
     socketPath: undefined,
     timeout: undefined,
     method: 'POST',
     path: '/graphql',
     _ended: true,
     res:
      IncomingMessage {
        _readableState: [ReadableState],
        readable: false,
        _events: [Object],
        _eventsCount: 3,
        _maxListeners: undefined,
        socket: [TLSSocket],
        connection: [TLSSocket],
        httpVersionMajor: 1,
        httpVersionMinor: 1,
        httpVersion: '1.1',
        complete: true,
        headers: [Object],
        rawHeaders: [Array],
        trailers: {},
        rawTrailers: [],
        aborted: false,
        upgrade: false,
        url: '',
        method: null,
        statusCode: 502,
        statusMessage: 'Bad Gateway',
        client: [TLSSocket],
        _consuming: false,
        _dumped: false,
        req: [Circular],
        responseUrl: 'https://api.github.com/graphql',
        redirects: [] },
     aborted: undefined,
     timeoutCb: null,
     upgradeOrConnect: false,
     parser: null,
     maxHeadersCount: null,
     _redirectable:
      Writable {
        _writableState: [WritableState],
        writable: true,
        _events: [Object],
        _eventsCount: 2,
        _maxListeners: undefined,
        _options: [Object],
        _redirectCount: 0,
        _redirects: [],
        _requestBodyLength: 2041,
        _requestBodyBuffers: [],
        _onNativeResponse: [Function],
        _currentRequest: [Circular],
        _currentUrl: 'https://api.github.com/graphql' },
     [Symbol(isCorked)]: false,
     [Symbol(outHeadersKey)]:
      { accept: [Array],
        'content-type': [Array],
        authorization: [Array],
        'user-agent': [Array],
        'content-length': [Array],
        host: [Array] } },
  response:
   { status: 502,
     statusText: 'Bad Gateway',
     headers:
      { server: 'GitHub.com',
        date: 'Thu, 25 Oct 2018 23:03:21 GMT',
        'content-type': 'application/json',
        'transfer-encoding': 'chunked',
        connection: 'close',
        'x-github-request-id': 'A4E6:7357:1033F7:223D81:5BD24BAF' },
     config:
      { adapter: [Function: httpAdapter],
        transformRequest: [Object],
        transformResponse: [Object],
        timeout: 0,
        xsrfCookieName: 'XSRF-TOKEN',
        xsrfHeaderName: 'X-XSRF-TOKEN',
        maxContentLength: -1,
        validateStatus: [Function: validateStatus],
        headers: [Object],
        method: 'post',
        url: 'https://api.github.com/graphql',
        data:
         '{"query":"\\nquery {\\n  organization(login: \\"MyBitFoundation\\"){\\n    repositories(first: 100){\\n      pageInfo{\\n        hasNextPage\\n      }\\n      edges{\\n        node{\\n          name,\\n          repositoryTopics(first: 10){\\n            edges{\\n              node{\\n                topic{\\n                  name\\n                }\\n              }\\n            }\\n          },\\n          issues(first: 100){\\n            pageInfo{\\n              hasNextPage\\n            }\\n            edges{\\n              node{\\n                createdAt,\\n                state,\\n                timeline(first: 10){\\n                  pageInfo{\\n                    hasNextPage\\n                  }\\n                  edges{\\n                    node{\\n                      ... on CrossReferencedEvent{\\n                        url\\n                        target\\n                        source{\\n                          ... on PullRequest{\\n                            url\\n                            state\\n                            author{\\n                              login\\n                            }\\n                          }\\n                        }\\n                      }\\n                    }\\n                    cursor\\n                  }\\n                },\\n                labels(first: 10){\\n                  edges{\\n                    node{\\n                      name,\\n                    },\\n                  }\\n                },\\n                url,\\n                title,\\n                number\\n                comments(first: 10){\\n                  pageInfo{\\n                    hasNextPage\\n                  }\\n                  edges{\\n                    node{\\n                      body,\\n                      author{\\n                        login\\n                      }\\n                    },\\n                    cursor\\n                  }\\n                }\\n              },\\n              cursor\\n            }\\n          }\\n        },\\n        cursor\\n      }\\n    }\\n  }\\n}"}' },
     request:
      ClientRequest {
        _events: [Object],
        _eventsCount: 6,
        _maxListeners: undefined,
        output: [],
        outputEncodings: [],
        outputCallbacks: [],
        outputSize: 0,
        writable: true,
        _last: true,
        chunkedEncoding: false,
        shouldKeepAlive: false,
        useChunkedEncodingByDefault: true,
        sendDate: false,
        _removedConnection: false,
        _removedContLen: false,
        _removedTE: false,
        _contentLength: null,
        _hasBody: true,
        _trailer: '',
        finished: true,
        _headerSent: true,
        socket: [TLSSocket],
        connection: [TLSSocket],
        _header:
         'POST /graphql HTTP/1.1\r\nAccept: application/json, text/plain, */*\r\nContent-Type: application/json;charset=utf-8\r\nAuthorization: token ELIDED\r\nUser-Agent: axios/0.18.0\r\nContent-Length: 2041\r\nHost: api.github.com\r\nConnection: close\r\n\r\n',
        _onPendingData: [Function: noopPendingOutput],
        agent: [Agent],
        socketPath: undefined,
        timeout: undefined,
        method: 'POST',
        path: '/graphql',
        _ended: true,
        res: [IncomingMessage],
        aborted: undefined,
        timeoutCb: null,
        upgradeOrConnect: false,
        parser: null,
        maxHeadersCount: null,
        _redirectable: [Writable],
        [Symbol(isCorked)]: false,
        [Symbol(outHeadersKey)]: [Object] },
     data: { data: 'null', errors: [Array] } } }
Fetching issues again in 5 seconds.
```

**Comment by bookmoons** [comment]
Same result on branches `develop` and `master`.

**Comment by bookmoons** [comment]
Thanks for looking at this guys. Will give it another try.

**Comment by bookmoons** [comment]
I confirm I'm able to get this running now, without errors.

**Comment by rd123myb** [comment]
Hey @bookmoons are you able to get on the MyBit gitter? (https://gitter.im/Mybit-Foundation/DDF)

**Comment by Speedsatoshi** [comment]
Probl here 503 temporarily Un available nginx/1.15.8 splution pls
